### ISSUE TITLE
Inconsistent report of inconsistent-return-statements when using exit / qui

### ISSUE THREAD

**Issue by GonozalIX** [pylint-dev/pylint#10508]
### Bug description

```python
# pylint: disable=consider-using-sys-exit, missing-module-docstring, missing-function-docstring
def func1(i):
    if i == 1:
        return i
    quit(1)

def func2(i):
    if i == 1:
        quit(1)
    return 1
```

### Configuration

```ini

```

### Command used

```shell
pre-commit run pylint --file=main.py
```

### Pylint output

```python
************* Module main
main.py:2:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

### Expected behavior

Pylint complains about inconsistent-return-statements only in func1 but not in func2.
Expected behaviour is that at least both functions are assessed the same.

If the quit statement should be used with a `return` or not is up to you :-) 

### Pylint version

```shell
pylint 3.3.8
python 3.12.11
```

### Additional dependencies

```python

```

**Comment by zenlyj** [comment]
Thanks for the report. If I understood your example correctly, the message should not be raised for `func1`, since there is no implicit `return None` when `quit()` is used.

While this is valid, it is worth noting that the message is not raised when using `sys.exit()`, which is preferred over `exit()` / `quit()`. Refer to [consider-using-sys-exit](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/consider-using-sys-exit.html).
```
import sys

def func1(i):
    if i == 1:
        return i
    sys.exit(1)  # `inconsistent-return-statements` not raised
```

**Comment by GonozalIX** [comment]
@zenlyj Exactly, and thanks for the summary.
